python/rmm/_lib/device_buffer.pyx

```diff
@@ -89,7 +89,7 @@ cdef class DeviceBuffer:
                     stream.c_synchronize()
 
         # Save a reference to the MR and stream used for allocation
-        self.mr = get_current_device_resource()
+        self.mr = get_current_device_resource().get()
```

Comment:
I think you can just do the following and avoid introducing a `get()` member function in `DeviceMemoryResource`:

```suggestion
        self.mr = get_current_device_resource().c_obj.get()
```